src/content-helper/dashboard-widget/provider.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import {
+	ContentHelperError,
+	ContentHelperErrorCode,
+} from '../../blocks/content-helper/content-helper-error';
+import { TopPostData } from './top-posts/model';
+import {
+	convertDateToString,
+	removeDaysFromDate,
+} from '../../blocks/shared/utils/date';
+
+/**
+ * The form of the response returned by the /stats/posts WordPress REST API
+ * endpoint.
+ */
+interface TopPostsApiResponse {
+	error?: Error;
+	data?: TopPostData[];
+}
+
+/**
+ * The form of the result returned by the getTopPosts() function.
+ */
+export interface GetTopPostsResult {
```

Comment:
Is it better to just return `TopPostData[]` and avoid this interface ...... for me it seems better because its simple.

Reply:
Done, and I've also added some code to handle the case of an empty data array.